modules/tinymce/src/plugins/table/main/ts/ui/TableDialog.ts

```diff
@@ -61,23 +70,24 @@ const applyDataToElement = (editor: Editor, tableElm: HTMLTableElement, data: Ta
     attrs.cellspacing = data.cellspacing;
   }
 
-  // TODO: this has to be reworked somehow, for example by introducing dedicated option, which
-  // will control whether child TD/THs should be processed or not
-  if (Options.shouldStyleWithCss(editor) && tableElm.children) {
+  // TINY-9837: Relevant data are applied on child TD/THs only if they have been modified since the previous dialog submission
+  if (shouldStyleWithCss && tableElm.children && (shouldApplyOnCell.border || shouldApplyOnCell.cellpadding || hasAdvancedTableTab && shouldApplyOnCell.bordercolor)) {
```

Comment:
This `if` statement feels a bit long. I am wondering if we could create the `cellStyles` object before the `if` statement and do something like this:
```
if (shouldStyleWithCss && tableElm.children && !Obj.isEmpty(cellStyles)) {
```

Reply:
I've made the code structure the following:
```ts
if (shouldStyleWithCss && tableElm.children) {
  const cellStyles = {};

  // conditionally add to cellStyles

  if (!Obj.isEmpty(cellStyles)) {
    // for loop to process cells
  }
}
```